src/cache.c

```diff
@@ -125,18 +124,19 @@ static void cache_evict_entries(git_cache *cache)
 		return;
 	}
 
+	i = 0;
```

Comment:
I might be completely wrong, but what about initializing this to `random(0, git_oidmap_size(cache->map) - evict_count)` ?

Reply:
Yeah, we sure could do this, but only because we do know the map's implementation details. We'd also need to wrap-around when reaching `git_oidmap_size`. But it sure keeps the original intention a lot better than my refactored code.

It just feels like the wrong data structure for the task at hand, at least in this specific cache eviction scenario. I'm not familiar with the code though, so I cannot say anything more to it. @ethomson, you've got any comment?

Reply:
Hrmf.  No?  If I'm reading this correctly, we walk through the cache evicting items in-order, the difference is the starting point.  It feels like if we're going to go to any lengths to change our eviction strategy that we should actually have a strategy based on age or something and not the ordering of the oid.  Starting at a random place in the list seems like it might be slightly better than always evicting starting at the beginning of the map (I can imagine the latter enabling some pathological cases) but they both sort of suck anyway.

(This might be what you were referring to @pks-t with the "wrong data structure" comment.)

Reply:
> It feels like if we're going to go to any lengths to change our eviction strategy that we should actually have a strategy based on age or something and not the ordering of the oid.

That's exactly my thought here, too. An LRU cache might make a lot more sense.It would require us to keep track of the cached objects twice, though: once in the LRU queue for cache eviction and once in the map for quick lookup via OID.

So the question is how to proceed right now. I don't feel like introducing the `random(0, git_oidmap_size(cache->map) - evict_count)` thing, as it again ties us to the map implementation details. I know it's unlikely that it'll change anyway soon, but it just doesn't feel right. If you say that'd be required to get this merged, then fine, so be it and I'll just use @tiennou's proposal. But in that case, we should definitely revisit and fix it soon.

Reply:
No, I don't think that this is significantly worse (in terms of cache eviction) than what's being replaced.  0, random, it's pretty mediocre either way.  I'll file an issue but let's merge this now.